src/hooks/useSettings.tsx

```diff
@@ -1,56 +1,64 @@
-import { useAsyncStorage } from '@react-native-async-storage/async-storage'
-import { useCallback, useEffect, useState } from 'react'
-import { SHAPESHIFT_URI } from 'react-native-dotenv'
-import { singletonHook } from 'react-singleton-hook'
-
-const useSettingsImpl = () => {
-  const [settings, setSettings] = useState<Record<string, unknown> | null>(null)
-  const { getItem, setItem, mergeItem } = useAsyncStorage('settings')
-
-  useEffect(() => {
-    if (!settings)
-      getItem()
-        .then(data => {
-          if (!data) {
-            setSettings({ SHAPESHIFT_URI })
-            return setItem(JSON.stringify({ SHAPESHIFT_URI }))
-          }
-          let parsed: Record<string, unknown> | null
-          try {
-            parsed = JSON.parse(data)
-          } catch (e) {
-            console.error('error parsing settings data: ', e)
-            parsed = {}
-          }
-          setSettings({ ...parsed, SHAPESHIFT_URI })
-          return
-        })
-        .catch(console.error)
-  }, [settings, getItem, setItem])
-
-  const setSetting = useCallback(
-    async (name: string, value: unknown) => {
+import { useQuery, useMutation, useQueryClient } from '@tanstack/react-query'
+import AsyncStorage from '@react-native-async-storage/async-storage'
+import { isValidUrl } from '../lib/navigationFilter'
+
+const SETTINGS_KEY = 'settings'
+
+const fetchSettings = async () => {
+  const data = await AsyncStorage.getItem(SETTINGS_KEY)
+  if (data) {
+    const parsed = JSON.parse(data)
+
+    if (parsed.EXPO_PUBLIC_SHAPESHIFT_URI && isValidUrl(parsed.EXPO_PUBLIC_SHAPESHIFT_URI)) {
```

Comment:
There is a case where you update the variable, but it's stored in the AsyncStorage, which is peristent between runs, so if you update an env variable, we need to check if it's equal!